client/src/app/components/gui-editor/gui-editor.component.ts

```diff
@@ -0,0 +1,93 @@
+import {
+    Component,
+    Input,
+    trigger,
+    style,
+    animate,
+    state,
+    transition
+} from "@angular/core";
+import {NgFor} from "@angular/common";
+import {FileModel} from "../../store/models/fs.models";
+import {PropertyInputComponent} from "../forms/inputs/property-input.component";
+import {GuiEditorService} from "./gui-editor.service";
+import {EditorSidebarComponent} from "./sidebar/editor-sidebar.component";
+import {PropertyPosition, VisibilityState} from "./animation.states";
+import {CommandLineComponent} from "./commandline/commandline.component";
+
+require("./gui-editor.component.scss");
+
+@Component({
+    selector: "gui-editor",
+    providers: [GuiEditorService],
+    directives: [NgFor, PropertyInputComponent, EditorSidebarComponent, CommandLineComponent],
+    animations: [
+        trigger("propertyPosition", [
+            state("left", style({
+                margin: '20px 0 0 0'
+            })),
+            state("center", style({
+                margin: '20px auto'
+            })),
+            transition("hidden => visible", animate("100ms ease-in")),
+            transition("visible => hidden", animate("100ms ease-out"))
+        ])
+    ],
+    template: `
+            <main>
+                <property-input @propertyPosition="propertyPosition"
```

Comment:
Why are you doing the whole property-input this way?
Why not make a standard form with standard form inputs?